compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/infer-effect-dependencies/reactive-ref-ternary.expect.md

```diff
@@ -0,0 +1,68 @@
+
+## Input
+
+```javascript
+// @inferEffectDependencies
+import {useRef, useEffect} from 'react';
+import {print, mutate} from 'shared-runtime';
+
+function Component({cond}) {
+  const arr = useRef([]);
+  const other = useRef([]);
+  const derived = cond ? arr : other;
+  // Avoid taking derived.current as a dependency
+  useEffect(() => {
+    mutate(derived.current);
+    print(derived.current);
+  });
+  return arr;
+}
+
+```
+
+## Code
+
+```javascript
+import { c as _c } from "react/compiler-runtime"; // @inferEffectDependencies
+import { useRef, useEffect } from "react";
+import { print, mutate } from "shared-runtime";
+
+function Component(t0) {
+  const $ = _c(4);
+  const { cond } = t0;
+  let t1;
+  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+    t1 = [];
+    $[0] = t1;
+  } else {
+    t1 = $[0];
+  }
+  const arr = useRef(t1);
+  let t2;
+  if ($[1] === Symbol.for("react.memo_cache_sentinel")) {
+    t2 = [];
+    $[1] = t2;
+  } else {
+    t2 = $[1];
+  }
+  const other = useRef(t2);
+  const derived = cond ? arr : other;
+  let t3;
+  if ($[2] !== derived) {
+    t3 = () => {
+      mutate(derived.current);
+      print(derived.current);
+    };
+    $[2] = derived;
+    $[3] = t3;
+  } else {
+    t3 = $[3];
+  }
+  useEffect(t3, [derived]);
```

Comment:
`derived` is now correctly added as a dependency. Prior to this PR, this deps array was empty ([playground link](https://0xeac7-forget.vercel.app/#N4Igzg9grgTgxgUxALhAejQAgAIEsB2AZgjAKKHFwAuAIggA4L4AmTcuCYAOvrgLb0IMKpmCYoYBACUEhADTjJ5SiIC+mQjAh9MAchgIAhtV0BuHv0HDRmejAJUFfKFUNUEmdZu16wAC0MDZgBaGCh8Kn4EMx4eQnDqXAh8TABhbUF8JioACjE4ZOZPAEpRHkxMAvwwEUCYTABeRWlZHIBtAF1i8xTK5JrMCCo-EkbmmUJ2rp6KqoHWewA3BCKmqqKAfkw6zGRB4ZIZzAxMAEFFiFwi1wBrAgBzTAXcZeYAOjhYAwjtsG2nhhMVj4OAAT3KzWUCGoORypQaAD4yr0Ks5XO4cs9Xh8vtluhCKnYHJiSC8VjiYN8qPjeqoaRUDFRYCk6j1VCBVEA))